src/Symfony/Component/Semaphore/Store/RedisStore.php

```diff
@@ -166,11 +166,7 @@ private function evaluate(string $script, string $resource, array $args): mixed
             return $this->redis->_instance($this->redis->_target($resource))->eval($script, array_merge([$resource], $args), 1);
         }
 
-        if ($this->redis instanceof \Predis\ClientInterface) {
-            return $this->redis->eval(...array_merge([$script, 1, $resource], $args));
-        }
-
-        throw new InvalidArgumentException(\sprintf('"%s()" expects being initialized with a Redis, RedisArray, RedisCluster or Predis\ClientInterface, "%s" given.', __METHOD__, get_debug_type($this->redis)));
```

Comment:
maybe we should keep this one: it can act as a defensive reminder the day we expand the types on the propety, if we do

Reply:
Suits me fine